### PR TITLE
Added spawner colours to `list_controllers` depending upon active or inactive

### LINKED ISSUE #687: [ros2_control CLI] `list_controller` should have colored output
## Background

Re-use colors defined in `spawner` script and make output on command line colored. Related to #686. Also correct aligment when using format-strings (initial work on this is done, but it should be optimized).

Files to check/update: https://github.com/ros-controls/ros2_control/blob/master/ros2controlcli/ros2controlcli/verb/list_controllers.py

## Instructions
Hi, this is a `good-first-issue` issue. This means we've worked to make it more legible to people who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**.

We're interested in helping you take the first step, and can answer questions and help you out along the way. Note that we're especially interested in contributions from underrepresented groups!

We know that creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this PR for someone new**, and looking through our general [bug](https://github.com/ros-controls/ros2_control/labels/bug) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 🗄️ **Create a local workspace** for making your changes and testing [following these instructions](https://docs.ros.org/en/galactic/Tutorials/Workspace/Creating-A-Workspace.html), for Step3 use "Download Source Code" section with [these instructions](https://ros-controls.github.io/control.ros.org/getting_started.html#compiling).

- [ ] 🍴 **Fork the repository** using the handy button at the top of the repository page and **clone** it into `~/ws_ros2_control/src/ros-controls/ros2_control`, [here is a guide that you can follow](https://guides.github.com/activities/forking/) (You will have to remove or empty the existing `ros2_control` folder before cloning your own fork)

- [ ] **Checkout a new branch** using `git checkout -b <branch_name>`

- [ ] 🤖 **Apply `pre-commit`** auto formatting, by running `pip3 install pre-commit` and running `pre-commit install` in the ros2_control repo.

- [ ] 💾 **Commit and Push** your changes

- [ ] 🔀 **Start a Pull Request** to request to merge your code into `master`. There are two ways that you can start a pull request:
1. If you are not familiar with GitHub or how to create a pull request, [here is a guide you can follow](https://guides.github.com/activities/hello-world/) on how GitHub works.

- [ ] 🏁 **Done** Ask in comments for a review :)

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it.

👥- **If someone seems stuck, offer them some help!**

### 🤔❓ Questions?

Don’t hesitate to ask questions or to get help if you feel like you are getting stuck. For example leave a comment below!
Furthermore, you find helpful resources here:
* [ROS2 Control Contribution Guide](https://ros-controls.github.io/control.ros.org/contributing.html)
* [ROS2 Tutorials](https://docs.ros.org/en/galactic/Tutorials.html)
* [ROS Answers](https://answers.ros.org/questions/)

**Good luck with your first issue!**

### PR DESCRIPTION
**Pull request by soham2560**
### This PR
Closes #687
- Made output for list_controllers coloured
  - Active controllers are highlighted with green `active`
  - Inactive controllers are highlighted with ~~red~~ cyan `inactive`
  - Unconfigured controllers are highlighted with yellow(warning) `unconfigured`

### Screenshots
image

### CODE REVIEW

**Comment by christophfroehlich** [comment]
Thanks for the nice addon. Should we add other colors for unconfigured state? It is printed in red now, if it is in any other state than active?

File: ros2controlcli/ros2controlcli/verb/list_controllers.py [hunk 1]
```diff
@@ -22,7 +23,8 @@
 
 
 def print_controller_state(c, args):
-    print(f"{c.name:20s}[{c.type:20s}] {c.state:10s}")
+    state_color = bcolors.OKGREEN if c.state == "active" else bcolors.FAIL
```
[the remark this answers is not in the document]
Comment (soham2560) [reply ros2controlcli/ros2controlcli/verb/list_controllers.py:26]: Yes, I think that makes much more sense. I'll make the necessary changes

**Review by olivier-stasse — CHANGES_REQUESTED** [review]

**Comment by soham2560** [comment]
I have changed the color scheme in [7be93b2](https://github.com/ros-controls/ros2_control/pull/1409/commits/7be93b2311e836509671ff37ebf24f9f1e3af43e). However 2 points
- I wasn't able to test this with a controller in the unconfigured state as I couldn't find a way to set that state purposefully, if you know of a way do let me know, I will test for that too
- We also haven't considered the finalized state, should I add that too?

File: ros2controlcli/ros2controlcli/verb/list_controllers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (soham2560) [reply ros2controlcli/ros2controlcli/verb/list_controllers.py]: refer https://github.com/ros-controls/ros2_control/pull/1409#issuecomment-1956676886

**Comment by saikishor** [comment]
> * I wasn't able to test this with a controller in the unconfigured state as I couldn't find a way to set that state purposefully, if you know of a way do let me know, I will test for that too

If you load a controller using `load_controller` service then it goes into unconfigured state
 
> * Current code also doesn't consider the finalized state, should I add that too?

I think internally it is not linked to any controller state, so I belive it should be good

**Comment by christophfroehlich** [comment]
Fyi, you can use rqt_controller_manager to set the state of a controller easily.
Have a look at this color scheme here:
https://github.com/ros-controls/ros2_control/blob/c4affe4bbfbb12b947acfc7b268f529724e0aae8/rqt_controller_manager/rqt_controller_manager/controller_manager.py#L87-L93
maybe we can use the same colors there as you have implemented it now?

**Review by fmauch — APPROVED** [review]
I like the improvement!

**Comment by soham2560** [comment]
Thanks for the tips regarding the controller manager @saikishor @christophfroehlich! Was able to test with the unconfigured state as well, I've added the result in the PR description, do have a look.

>maybe we can use the same colors there as you have implemented it now?

That sounds like a good idea, I can have a look at that if you want

**Review by olivier-stasse — APPROVED** [review]

**Review by saikishor — APPROVED** [review]
LGTM

**Review by destogl — APPROVED** [review]

**Review by bmagyar — APPROVED** [review]

**Pull request merged**